src/js/iterator.js

```diff
@@ -234,6 +244,7 @@ export default class Iterator {
     while (true) {
       let comparison = compare(position, this.currentNodePosition)
 
+      this.cacheNodePosition()
```

Comment:
Sorry for the late comment... is there any reason you can't call `cacheNodePosition` within `setNode`? Pretty sure that code path is exercised every time we move. I guess one reason is there are certain operations where we don't want to cache the node position when setting the node.